src/joystick/linux/SDL_sysjoystick.c

```diff
@@ -282,21 +282,32 @@ static int IsJoystick(const char *path, int fd, char **name_return, Uint16 *vend
 
     SDL_zero(inpid);
 #ifdef SDL_USE_LIBUDEV
+    /* Opening input devices can generate synchronous device I/O, so avoid it if we can */
     SDL_UDEV_GetProductInfo(path, &inpid.vendor, &inpid.product, &inpid.version, &class);
+    if (!(class & SDL_UDEV_DEVICE_JOYSTICK)) {
```

Comment:
This should not early out if SDL_UDEV_GetProductInfo() fails, the same holds true for the accelerometer check below.

Reply:
Thanks for spotting that. :+1: 

I have pushed a new version that fixes those two and the steam virtual games pad one too.

Sorry for the long delay.